core/node/groups.go

```diff
@@ -316,8 +318,20 @@ func IPFS(ctx context.Context, bcfg *BuildCfg) fx.Option {
 		return bcfgOpts // error
 	}
 
-	// TEMP: setting global sharding switch here
-	uio.UseHAMTSharding = cfg.Experimental.ShardingEnabled
+	// Auto-sharding settings
+	shardSizeString := cfg.Internal.UnixFSShardingSizeThreshold.WithDefault("256kiB")
+	shardSizeInt, err := humanize.ParseBytes(shardSizeString)
+	if err != nil {
+		return fx.Error(err)
+	}
+	uio.HAMTShardingSize = int(shardSizeInt)
+
+	// Migrate users of deprecated Experimental.ShardingEnabled flag
+	if cfg.Experimental.ShardingEnabled {
+		logger.Fatal("The `Experimental.ShardingEnabled` field is no longer used, please remove it from the config.\n" +
+			"go-ipfs now automatically shards when directory block is bigger than  `" + shardSizeString + "`.\n" +
+			"If you need to restore the old behavior (sharding everything) set `Internal.UnixFSShardingSizeThreshold` to `1B`.\n")
+	}
```

Comment:
:information_source:  This provides a migration path for existing users that have experiment enabled.
(I also updated `docs/experimental-features.md`)

Reply:
Thanks. This seems like the best we can get.

There's no way to warn anybody who explicitly wanted sharding to stay off since that flag didn't have no way to denote "unset" which is what we're moving towards now.